mesonbuild/modules/i18n.py

```diff
@@ -64,6 +64,7 @@ def _get_data_dirs(state, dirs):
         return [path.join(src_dir, d) for d in dirs]
 
     @FeatureNew('i18n.merge_file', '0.37.0')
+    @FeatureNewKwargs('i18n.gettext', '0.38.0', ['build_by_default'])
```

Comment:
I don't think this is the right approach because `merge_file` returns a `CustomTarget`, and `build_by_default` is actually an argument of `CustomTarget`, which merge_file just proxies. There's also a mess of arguments in the permitted kwargs for `merge_file` that it doesn't actually support. I think the right thing to do here is set the permittedKwargs to:
```python
build.CustomTarget.known_kwargs | {'data_dirs', 'po_dirs', 'type'}
```

That way we're not duplicating the same logic between two different places, and the i18n module will be automatically updated whenever the CustomTarget class is.

Reply:
Thanks. Updated.